token/program-2022/src/offchain.rs

```diff
@@ -11,58 +11,6 @@ use {
     std::future::Future,
 };
 
-/// Offchain helper to get all additional required account metas for a checked
-/// transfer
-///
-/// To be client-agnostic and to avoid pulling in the full solana-sdk, this
-/// simply takes a function that will return its data as `Future<Vec<u8>>` for
-/// the given address. Can be called in the following way:
-///
-/// ```rust,ignore
-/// use futures_util::TryFutureExt;
-/// use solana_client::nonblocking::rpc_client::RpcClient;
-/// use solana_program::pubkey::Pubkey;
-///
-/// let mint = Pubkey::new_unique();
-/// let client = RpcClient::new_mock("succeeds".to_string());
-/// let mut account_metas = vec![];
-///
-/// get_extra_transfer_account_metas(
-///     &mut account_metas,
-///     |address| self.client.get_account(&address).map_ok(|opt| opt.map(|acc| acc.data)),
-///     &mint,
-/// ).await?;
-/// ```
-#[deprecated(
-    since = "1.1.0",
-    note = "Please use `create_transfer_checked_instruction_with_extra_metas` instead"
-)]
-pub async fn resolve_extra_transfer_account_metas<F, Fut>(
```

Comment:
Oh darn, sorry, I forgot about this helper here. We'll need to do a new major version release of token-2022.

I think that's fine, considering we'll also formally enable the token groups with 1.18, but we should be aware of that.

Reply:
Do you want to do that now or wait a little bit?

Reply:
we can hold off on releasing token-2022 v2 until Solana v1.18, but feel free to get all the other crates out!

Reply:
The client would be on the deprecated offchain helper until we released Token2022. Unless we temporarily injected the offchain helper from SPL Transfer Hook temporarily, which we could do.

Reply:
The client would just have to manually do this:

https://github.com/solana-labs/solana-program-library/blob/23916b24700cd222ad2397e9de8162ad6980e2e8/token/program-2022/src/offchain.rs#L104-L134

And anyone attempting a CPI to Token2022 with extra metas would need to do this setup manually as well:

https://github.com/solana-labs/solana-program-library/blob/23916b24700cd222ad2397e9de8162ad6980e2e8/token/program-2022/src/onchain.rs#L31-L79

Reply:
So CPIs would be a little boned until that comes out. But the client we can sneak that in.

Reply:
gotcha, we can get token-2022 v2 out sooner then, but we'll just need to clarify in the release notes that it's not going on-chain